logstash-core/lib/logstash/agent.rb

```diff
@@ -88,9 +88,9 @@ def execute
   # @param pipeline_id [String] pipeline string identifier
   # @param settings [Hash] settings that will be passed when creating the pipeline.
   #   keys should be symbols such as :pipeline_workers and :pipeline_batch_delay
-  def register_pipeline(pipeline_id, settings = @settings)
+  def register_pipeline(settings = @settings)
```

Comment:
Can we remove the fact that we default to an instance variable here? It feel a bit weird to me and could result in unexpected behavior.